pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -48,9 +48,12 @@ type caOptions struct {
 	ExternalCAType   ra.CaExternalType
 	ExternalCASigner string
 	// domain to use in SPIFFE identity URLs
-	TrustDomain      string
-	Namespace        string
-	Authenticators   []security.Authenticator
+	TrustDomain    string
+	Namespace      string
+	Authenticators []security.Authenticator
+	// CertSignerDomain is based on the CERT_SIGNER_DOMAIN env variable
+	// It is combined with the user-supplied name to form the K8S signer for the RA:
+	// CertSignerDomain + "/" + CertSigner ( which is passed via injector CERT_SIGNER)
```

Comment:
Yes, we should consider deprecating this stuff

Reply:
Will do another round of cleanup - I found more uses of the deprecated signer - the k8s.io/ prefix for example. 

I would also make the loading of the roots more consistent in a separate PR - allow MeshCA roots for all signers, allow the mounted CA roots for all signers - domain signer prefix is the only case where MeshCA is required, but for everything else we can use either MeshCA or mounted CA.

I would give top priority to the mounted file, even if DNS istio certs are not explicitly configured.